From c06bb87cf6575b2b6880cf18c4e18a746841366d Mon Sep 17 00:00:00 2001
From: Harvin Almeida <harvin.almeida@analog.com>
Date: Tue, 17 Aug 2021 20:41:34 +0000
Subject: [PATCH] Update physical address IOCTL for Kernel 5.2

---
 drivers/misc/adi/sram_mmap.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/misc/adi/sram_mmap.c b/drivers/misc/adi/sram_mmap.c
index 2b468f0663df..a1328d0c58dc 100644
--- a/drivers/misc/adi/sram_mmap.c
+++ b/drivers/misc/adi/sram_mmap.c
@@ -118,7 +118,7 @@ static long sram_ioctl(struct file *file,unsigned int cmd, unsigned long arg)
 	int err=0;
 
 	if(_IOC_DIR(cmd) & _IOC_READ)
-		err = !access_ok(VERIFY_WRITE, (void __user *)arg, _IOC_SIZE(cmd));
+		err = !access_ok((void __user *)arg, _IOC_SIZE(cmd));
 
 	switch (cmd)
 	{
-- 
2.17.1

