From 96e9146014ab8b23b0248b53e71b2ca2ea39c3a0 Mon Sep 17 00:00:00 2001
From: Harvin Almeida <harvin.almeida@analog.com>
Date: Tue, 2 Mar 2021 20:16:53 +0000
Subject: [PATCH] Add USB boot to u-boot

---
 include/configs/sc_adi_common.h | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/include/configs/sc_adi_common.h b/include/configs/sc_adi_common.h
index ab9947b..4d8bb16 100644
--- a/include/configs/sc_adi_common.h
+++ b/include/configs/sc_adi_common.h
@@ -105,6 +105,7 @@
 
 #define CONFIG_BOOTARGS_ROOT_NAND "/dev/mtdblock2 rw"
 #define CONFIG_BOOTARGS_ROOT_SDCARD    "/dev/mmcblk0p1 rw"
+#define CONFIG_BOOTARGS_ROOT_USB "/dev/sda1 rw"
 
 #define CONFIG_BOOTARGS_SDCARD	\
 	"root=" CONFIG_BOOTARGS_ROOT_SDCARD " " \
@@ -116,6 +117,16 @@
 			__stringify(CONFIG_BAUDRATE) " "\
 	"mem=" CONFIG_LINUX_MEMSIZE
 
+#define CONFIG_BOOTARGS_USB \
+	"root=" CONFIG_BOOTARGS_ROOT_USB " rootwait " \
+	"rootfstype=ext3 " \
+	"clkin_hz=" __stringify(CONFIG_CLKIN_HZ) " " \
+	CONFIG_BOOTARGS_VIDEO \
+	"earlyprintk=serial,uart0,57600 " \
+	"console=ttySC" __stringify(CONFIG_UART_CONSOLE) "," \
+			__stringify(CONFIG_BAUDRATE) " "\
+	"mem=" CONFIG_LINUX_MEMSIZE
+
 #define CONFIG_BOOTARGS_NFS	\
 	"root=/dev/nfs rw " \
 	"nfsroot=${serverip}:${rootpath},tcp,nfsvers=3 " \
@@ -180,6 +191,7 @@
 	"dtbfile=" CONFIG_DTBNAME "\0" \
 	"dtbaddr=" CONFIG_DTBLOADADDR "\0" \
 	"sdcardargs=set bootargs " CONFIG_BOOTARGS_SDCARD "\0" \
+	"usbargs=set bootargs " CONFIG_BOOTARGS_USB "\0" \
 	"ramargs=set bootargs " CONFIG_BOOTARGS "\0" \
 	"nfsargs=set bootargs " CONFIG_BOOTARGS_NFS "\0" \
 	"ramboot_emmc=" \
@@ -219,6 +231,15 @@
 		"bootz ${loadaddr} - ${dtbaddr}" \
 		"\0" \
 	\
+	"usbboot=" \
+		"usb start;" \
+		"ext2load usb 0:1 ${loadaddr} ${ramfile};" \
+		"ext2load usb 0:1 ${dtbaddr} ${dtbfile};" \
+		"run usbargs;" \
+		"run addip;" \
+		"bootz ${loadaddr} - ${dtbaddr}" \
+		"\0" \
+	\
 	"nfsfile=zImage\0" \
 	"nfsboot=" \
 		"tftp ${loadaddr} ${nfsfile};" \
