From 5074d28a5b1aa0c500df5469dde259caf5302871 Mon Sep 17 00:00:00 2001
From: Harvin Almeida <harvin.almeida@analog.com>
Date: Tue, 5 Nov 2024 13:59:28 -0600
Subject: [PATCH] 312 IOCTL to get last used physical address

---
 drivers/misc/adi/sram_mmap.c | 27 +++++++++++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/drivers/misc/adi/sram_mmap.c b/drivers/misc/adi/sram_mmap.c
index a6fdc41035f6..c50f92824318 100644
--- a/drivers/misc/adi/sram_mmap.c
+++ b/drivers/misc/adi/sram_mmap.c
@@ -22,8 +22,12 @@
 #include <linux/genalloc.h>
 #include <linux/miscdevice.h>
 #include <linux/platform_device.h>
+#include <linux/ioctl.h>
 
 #define SRAM_MMAP_DRV_NAME		"sram_mmap"
+#define PHYSADDR _IOR(0x6B,0x01,unsigned long)
+
+unsigned long paddr = 0;
 
 #define MAX_SRAM_REGIONS 4
 struct sram_region_info {
@@ -115,14 +119,37 @@ static int sram_mmap(struct file *fp, struct vm_area_struct *vma)
 		}
 	}
 
+	paddr = (size_t) sram->rmem->base + vma->vm_pgoff * PAGE_SIZE;
+
 	dev_info(sram->dev, "Mapped 0x%zx : 0x%zx successfully\n",
 		(size_t) (sram->rmem->base + vma->vm_pgoff * PAGE_SIZE),
 		(size_t) (sram->rmem->base + vma->vm_pgoff * PAGE_SIZE + sram_size));
 	return 0;
 }
 
+static long sram_ioctl(struct file *file,unsigned int cmd, unsigned long arg)
+{
+	int err=0;
+
+	if(_IOC_DIR(cmd) & _IOC_READ)
+		err = !access_ok((void __user *)arg, _IOC_SIZE(cmd));
+
+	switch (cmd)
+	{
+	case PHYSADDR:
+		put_user(paddr,(unsigned long __user *)arg);
+		break;
+
+	default:
+		break;
+	}
+
+	return err;
+}
+
 static const struct file_operations sram_fops = {
 	.mmap		= sram_mmap,
+	.unlocked_ioctl = sram_ioctl,
 };
 
 static const struct of_device_id adi_sram_mmap_of_match[] = {
-- 
2.25.1

