diff --git a/arch/arm64/boot/dts/adi/sc598-som-ezkit.dts b/arch/arm64/boot/dts/adi/sc598-som-ezkit.dts
index abd954fb32b4..f0846778acfa 100644
--- a/arch/arm64/boot/dts/adi/sc598-som-ezkit.dts
+++ b/arch/arm64/boot/dts/adi/sc598-som-ezkit.dts
@@ -141,6 +141,13 @@
 	/* adi,average; */
 };
 
+&gptimer3 {
+	/* enable PB_03 to work as TIMER0_TMR3 */
+	pinctrl-names = "default";
+	pinctrl-0 = <&extclkin_default>;
+	status = "okay";
+};
+
 &uart0 {
 	/* enable-pin = <&ssw0 5 GPIO_ACTIVE_LOW>;	 UART0_EN */
 	/* hwflow-en-pin = <&ssw0 6 GPIO_ACTIVE_LOW>;  UART0_FLOW_EN */
@@ -598,6 +605,11 @@
 };
 
 &pinctrl0 {
+	extclkin_default: extclkin_default_pins {
+		pins {
+			pinmux = <ADI_ADSP_PINMUX('B', 3, ADI_ADSP_PINFUNC_ALT0)>;
+		};
+	};
 	uart0_default: uart0_default_pins {
 		pins {
 			pinmux = <ADI_ADSP_PINMUX('A', 6, ADI_ADSP_PINFUNC_ALT1)>,
