diff --git a/drivers/remoteproc/remoteproc_core.c b/drivers/remoteproc/remoteproc_core.c
index ce92ae227aa1..e406e38801a5 100644
--- a/drivers/remoteproc/remoteproc_core.c
+++ b/drivers/remoteproc/remoteproc_core.c
@@ -2029,7 +2029,7 @@ struct rproc *rproc_alloc(struct device *dev, const char *name,
 	rproc->firmware = p;
 	rproc->name = name;
 	rproc->priv = &rproc[1];
-	rproc->auto_boot = true;
+	rproc->auto_boot = false;
 
 	device_initialize(&rproc->dev);
 	rproc->dev.parent = dev;
