From 6ed78b8cceff05e8ac53f5d708216d6e9a7cbaf4 Mon Sep 17 00:00:00 2001
From: Harvin Almeida <harvin.almeida@analog.com>
Date: Fri, 15 May 2020 18:40:50 +0000
Subject: [PATCH] hrtimer-support

---
 arch/arm/mach-sc58x/core.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/arch/arm/mach-sc58x/core.c b/arch/arm/mach-sc58x/core.c
index 2dde56e24427..250e38439a1d 100644
--- a/arch/arm/mach-sc58x/core.c
+++ b/arch/arm/mach-sc58x/core.c
@@ -545,7 +545,6 @@ static int gptmr_set_state_oneshot(struct clock_event_device *evt)
 {
 	int id = timer_event->id;
 
-	while(1);
 	disable_gptimers(1 << id);
 	set_gptimer_config(timer_event, TIMER_OUT_DIS | TIMER_MODE_PWM
 			| TIMER_PULSE_HI | TIMER_IRQ_WID_DLY);
@@ -595,7 +594,7 @@ static struct clock_event_device clockevent_gptmr = {
 	.name           = "sc58x_gptimer0",
 	.rating         = 300,
 	.shift          = 32,
-	.features       = CLOCK_EVT_FEAT_PERIODIC,
+	.features       = CLOCK_EVT_FEAT_PERIODIC | CLOCK_EVT_FEAT_ONESHOT,
 	.set_next_event = gptmr_set_next_event,
 	.set_state_periodic = gptmr_set_state_periodic,
 	.set_state_oneshot = gptmr_set_state_oneshot,
