From 3d3a810d24c51f1712b619a7991f20e4e5dd7fed Mon Sep 17 00:00:00 2001
From: Harvin Almeida <harvin.almeida@analog.com>
Date: Tue, 2 Mar 2021 20:17:44 +0000
Subject: [PATCH] Set usbboot as deafult on sc573-ezkit

---
 include/configs/sc573-ezkit.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/sc573-ezkit.h b/include/configs/sc573-ezkit.h
index 13e6a9d..ac6ff1d 100644
--- a/include/configs/sc573-ezkit.h
+++ b/include/configs/sc573-ezkit.h
@@ -167,7 +167,7 @@
 #define CONFIG_DWMMC
 #define CONFIG_BOUNCE_BUFFER
 
-#define CONFIG_BOOTCOMMAND	"run ramboot"
+#define CONFIG_BOOTCOMMAND	"run usbboot"
 #define INITRAMADDR "0x85000000"
 
 #include <configs/sc_adi_common.h>
